clients/go/client.go

```diff
@@ -29,6 +33,13 @@ type Client interface {
 	// Refreshes proto definitions if parsed message has unknown fields and parses the message again.
 	// Returns ErrNotFound error if given class name is not found.
 	ParseWithRefresh(string, []byte) (protoreflect.ProtoMessage, error)
+	// Serialize serializes data to bytes given fully qualified name of proto message.
+	// Returns ErrNotFound error if given class name is not found
+	Serialize(string, interface{}) ([]byte, error)
+	// Serialize serializes data to bytes given fully qualified name of proto message.
+	// Refreshes proto definitions if parsed message has unknown fields and parses the message again.
+	// Returns ErrNotFound error if given class name is not found.
+	SerializeWithRefresh(string, interface{}) ([]byte, error)
```

Comment:
SerializeWithRefresh Not needed I guess...Since data is with producer and it knows which fields are present....Not really needed for serialization.

Reply:
oh what if there are new fields in the proto and it's not being sent? should we concern about that?

Reply:
if producer doesn't have a data for corresponding field we can left it out I guess. Maybe we need to check if all passed data encoded properly to protobuf message.

Reply:
hi @harikrishnakanchi! so for serializing, there will be no case when in the data there's a new field or new enum type being sent but the descriptor is not refreshed yet?

Reply:
> hi @harikrishnakanchi! so for serializing, there will be no case when in the data there's a new field or new enum type being sent but the descriptor is not refreshed yet?

+1

> Maybe we need to check if all passed data encoded properly to protobuf message.

modified the test to satisfy this